payments-core/api/payments-core.api

```diff
@@ -14,6 +14,40 @@ public final class com/stripe/android/BuildConfig {
 	public fun <init> ()V
 }
 
+public abstract interface class com/stripe/android/CreateIntentCallback : com/stripe/android/AbsCreateIntentCallback {
+	public abstract fun onCreateIntent (Ljava/lang/String;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+}
+
+public abstract interface class com/stripe/android/CreateIntentCallbackForServerSideConfirmation : com/stripe/android/AbsCreateIntentCallback {
+	public abstract fun onCreateIntent (Ljava/lang/String;ZLkotlin/coroutines/Continuation;)Ljava/lang/Object;
+}
+
+public abstract interface class com/stripe/android/CreateIntentResult {
+}
+
+public final class com/stripe/android/CreateIntentResult$Failure : com/stripe/android/CreateIntentResult {
```

Comment:
Should we create factories for these instead? Then we can keep the classes private.

Reply:
Talked offline, we'll explore this after private beta.